src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -47,37 +47,45 @@ public function supports(Request $request, ArgumentMetadata $argument): bool
             return false;
         }
 
-        $options = $this->getOptions($argument);
+        $options = $this->getMapEntityAttribute($argument);
         if (!$options->class || $options->disabled) {
             return false;
         }
 
+        if (null === $options->expr && !($options->mapping || $options->exclude) && null === $this->getIdentifier($request, $options, $argument->getName())) {
```

Comment:
this is broken in case `$options->mapping === null`, as this case creates an automatic mapping based in `getCriteria`, which might find something even if `getIdentifier` returns `null`. You need to distinguish the case of the empty array or null.

Reply:
if getIdentifier returns null we don't call findOneBy since #47242, or did I missunderstand you?

Reply:
oh, indeed. We search by criteria when it returns `false`, not `null`.